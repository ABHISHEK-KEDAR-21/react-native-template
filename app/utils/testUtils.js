import React from 'react'
import { render } from '@testing-library/react-native'
import { Provider } from 'react-redux'
import ConnectedLanguageProvider from 'app/containers/LanguageProvider'
import { translationMessages } from '../i18n'
import createStore from '../rootReducer'
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
})

export const renderProvider = children => {
  const { store } = createStore()
  return render(
    <Provider store={store}>
      <ConnectedLanguageProvider messages={translationMessages}>
        {children}
      </ConnectedLanguageProvider>
    </Provider>
  )
}
