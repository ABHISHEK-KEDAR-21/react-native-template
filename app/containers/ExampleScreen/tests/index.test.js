/**
 *
 * Tests for ExampleScreen
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { renderProvider } from '../../../utils/testUtils'
import { ExampleScreenTest } from '../index'

describe('<ExampleScreen />', () => {
  let submitSpy

  beforeAll(() => {
    submitSpy = jest.fn()
  })

  it('Should render and match the snapshot', () => {
    const { container } = renderProvider(
      <ExampleScreenTest fetchUser={submitSpy} />
    )
    expect(container.children[0]).toMatchSnapshot()
  })

  it('should fetch the user data on mount', () => {
    renderProvider(<ExampleScreenTest fetchUser={submitSpy} />)
    expect(submitSpy).toHaveBeenCalled()
  })
  it('should render ActivityIndicator if userIsLoading is true', () => {
    const { getByTestId } = renderProvider(
      <ExampleScreenTest fetchUser={submitSpy} userIsLoading />
    )

    // expect(getByTestId('loader')).toExist()
    expect(getByTestId('loader').type).toBe('ActivityIndicator')
    expect(submitSpy).toHaveBeenCalled()
  })

  it('should not render ActivityIndicator if userIsLoading is false, should instead render exampleContainerContent', () => {
    const { getByTestId } = renderProvider(
      <ExampleScreenTest fetchUser={submitSpy} userIsLoading={false} />
    )
    expect(getByTestId('exampleContainerContent').type).toBe('View')
    expect(submitSpy).toHaveBeenCalled()
  })
})
