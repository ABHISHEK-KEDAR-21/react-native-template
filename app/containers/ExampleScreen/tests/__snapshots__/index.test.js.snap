// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExampleScreen /> Should render and match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "marginBottom": 30,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 30,
      },
    ]
  }
>
  <View
    testID="exampleContainerContent"
  >
    <View
      style={
        Array [
          Object {
            "height": 150,
            "marginBottom": 25,
            "width": "100%",
          },
        ]
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../app/assets/images/logo.png",
          }
        }
        style={
          Array [
            Object {
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "fontSize": 17,
            "marginBottom": 5,
            "textAlign": "center",
          },
        ]
      }
    >
      To get started, edit App.js
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontSize": 17,
            "fontStyle": "italic",
            "marginBottom": 5,
            "textAlign": "center",
          },
        ]
      }
    >
      Press Cmd+R to reload,
Cmd+D or shake for dev menu.
    </Text>
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 17,
              "marginBottom": 5,
              "textAlign": "center",
            },
          ]
        }
      >
        <Text>
          character loves Wednesday
        </Text>
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 17,
              "marginBottom": 5,
              "textAlign": "center",
            },
          ]
        }
      >
        <Text>
          because
        </Text>
      </Text>
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Array [
            Object {
              "height": 80,
              "marginBottom": 0,
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": 0,
              "width": 80,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "fontSize": 17,
              "marginBottom": 5,
              "textAlign": "center",
            },
          ]
        }
      />
    </View>
    <Button
      style={
        Array [
          Object {
            "marginTop": 40,
          },
        ]
      }
      title="Refresh"
    />
  </View>
</View>
`;
